File: pay/paypayapi.py
from flask import Flask, request, jsonify
import requests

app = Flask(__name__)

@app.route('/pay', methods=['POST'])
def pay():
    data = request.json
    user_id = data['user_id']
    store_id = data['store_id']
    menue_id = data['menue_id']
    reserv_time = data['reserv_time']
    reserv_cnt = data['reserv_cnt']
    is_recipt = data['is_recipt']
    total_amount = data['total_amount']

    # PayPay APIへのリクエスト
    paypay_api_url = "https://api.paypay.ne.jp/v2/payments"
    headers = {
        "Content-Type": "application/json",
        "Authorization": "Bearer YOUR_PAYPAY_API_KEY"
    }
    payload = {
        "merchantPaymentId": f"{user_id}-{store_id}-{menue_id}",
        "amount": {
            "amount": total_amount,
            "currency": "JPY"
        },
        "orderDescription": "Bento reservation payment",
        "orderItems": [
            {
                "name": "Bento",
                "quantity": reserv_cnt,
                "unit_price": {
                    "amount": total_amount / reserv_cnt,
                    "currency": "JPY"
                }
            }
        ],
        "userAgent": request.headers.get('User-Agent')
    }

    response = requests.post(paypay_api_url, headers=headers, json=payload)
    
    if response.status_code == 200:
        return jsonify({"message": "Payment successful"})
    else:
        return jsonify({"error": "Payment failed"}), response.status_code

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5000)

File: pay/paypayapi.js
'use strict';
const PAYPAY = require('@paypayopa/paypayopa-sdk-node');
const express = require('express');
const bodyParse = require('body-parser');

const app = express();
app.use(bodyParser.json());

console.log(process.argv);

const paypayClient = new paypay.Client({
    clientId: process.env.PAYPAY_CLIENT_ID,
    clientSecret: process.env.PAYPAY_CLIENT_SECRET,
    merchantId: process.env.PAYPAY_MERCHANT_ID,
    productionMode: false // true for production
});

const { reserv_id, user_id, store_id, menue_id, reserv_time, reserv_cnt, is_recipt, total_amount } = req.body;

// 支払リクエストを作成
const payload = {
    merchantPaymentId: `test_merchant_payment_id_${reserv_id}`,
    amount: {
        amount: total_amount,
        currency: 'JPY'
    },
    requestedAt: new Date().getTime(),
    redirectUrl: 'https://paypay.ne.jp/',
    redirectType: 'WEB_LINK'
};

try {
    const result = await paypayClient.QRCodeCreate(payload);
    if (result.data) {
        res.json({ url: result.data.url });
    } else {
        res.status(500).json({ error: 'Payment creation failed' });
    }
} catch (error) {
    res.status(500).json({ error: error.message });
}
